### PR TITLE
refactor: switch to eslint v9, eslint-config-eslint v10

### PR DESCRIPTION
**Pull request by mdjermanovic**
Switches to using eslint v9 and eslint-config-eslint v10 to lint this project.

Marked as `refactor` as it has changes in the production code.

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -23,11 +23,7 @@ export default [
         files: ["tools/**"],
         rules: {
             "no-console": "off",
-            "n/no-process-exit": "off",
-            "n/no-unsupported-features/es-syntax": ["error", {
-                version: ">=16.0.0",
-                ignores: ["modules"]
-            }]
```
Comment (mdjermanovic) [inline eslint.config.js:30]: This override is apparently no longer need with the latest version of eslint-plugin-n.

**Review by aladdin-add — APPROVED** [review]

File: espree.js [hunk 2]
```diff
@@ -160,7 +157,7 @@ export const Syntax = (function() {
     }
 
     for (key in VisitorKeys) {
-        if (Object.hasOwnProperty.call(VisitorKeys, key)) {
+        if (Object.hasOwn(VisitorKeys, key)) {
```
Comment (mdjermanovic) [inline espree.js:160]: This was reported by `prefer-object-has-own`.

File: lib/options.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ const SUPPORTED_VERSIONS = [
  * @returns {number} The latest ECMAScript version.
  */
 export function getLatestEcmaVersion() {
-    return SUPPORTED_VERSIONS[SUPPORTED_VERSIONS.length - 1];
+    return SUPPORTED_VERSIONS.at(-1);
```
Comment (mdjermanovic) [inline lib/options.js:30]: This was reported by `unicorn/prefer-at`.

File: tools/update-tests.js [hunk 4]
```diff
@@ -70,14 +70,13 @@ const testFiles = getTestFilenames(FIXTURES_DIR),
...
-    let result = tester.getExpectedResult(code, {
+    const result = tester.getExpectedResult(code, {
         loc: true,
         range: true,
         tokens: true
     });
 
     JSON.stringify(result).to(testResultFilename);
-    result = null;
 });
```
Comment (mdjermanovic) [inline tools/update-tests.js:80]: This was reported by `no-useless-assignment`.

**Pull request merged**